wire-ios-sync-engine/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoder.m

```diff
@@ -330,10 +327,6 @@ - (NSUUID *)nextUUIDFromResponse:(ZMTransportResponse *)response forListPaginato
 
     NSUUID *latestEventId = [self processUpdateEventsAndReturnLastNotificationIDFromResponse:response];
 
-    if (!self.listPaginator.hasMoreToFetch) {
-        [self.previouslyReceivedEventIDsCollection discardListOfAlreadyReceivedPushEventIDs];
-    }
```

Comment:
this is where we may discard already received events before decrypting and storing them because this happens in the `processUpdateEventsAndReturnLastNotificationIDFromResponse:` method above but this is asynchronous